creator/data_templates/packaging.py

```diff
@@ -0,0 +1,272 @@
+from typing import Union, List, Optional
+import os
+from tempfile import TemporaryDirectory
+from zipfile import ZipFile
+from io import BytesIO
+from pprint import pformat
+
+import pandas
+
+from creator.studies.models import Study
+from creator.data_templates.models import TemplateVersion
+
+MAX_SHEET_NAME_LEN = 31
+
+PACKAGE_INFO = (
+    """
+    The Fields file describes all of the fields or columns in the template
+    (name, data type, required or not, etc.) while the blank Template file
+    is what data submitters will fill out and then submit to their
+    organization for ingestion.
+    """
+)
+
+
+def make_sheet_name(template_name: str, type_: str):
+    """
+    Helper to create a Excel sheet name from the template name and it's type
+    (e.g. Fields or Template)
+
+    Create a sheet_name by putting together the template name and type. If
+    its > the max char limit for sheet names, truncate the template name
+    until the sheet_name has a valid length
+    """
+    suffix = f" - {type_}"
+    sheet_name = f"{template_name}{suffix}"
+    # Truncate template name to avoid Excel writer exception
+    if len(sheet_name) > MAX_SHEET_NAME_LEN:
+        sheet_name = template_name[
+            0:(MAX_SHEET_NAME_LEN - len(sheet_name))
+        ] + suffix
+    return sheet_name
```

Comment:
```suggestion
    return f"{template_name[: MAX_SHEET_NAME_LEN]}{suffix}"
```

Reply:
Your way works! I forgot that the stop index when list slicing can actually be greater than the length of the list. I think what I have works bc the stop index will always be negative. 

But I like your approach better. Its cleaner and I don't have to check for the len of sheet name